### PR TITLE
[#99042646] add queue_depth and batch_size

### PR DESCRIPTION
**Pull request by james-barnard**
This PR expands the queue metric tracker to include queue_depth and batch_size.

queue_depth: the number of remaining devices with messages in the queue
batch_size: the number of messages retrieved for a single device
N: the number of tradesman fetches that will be retrieved before a metric will be reported

queue_depth and batch_size are averaged over N fetches

### CODE REVIEW

File: lib/worker_roulette/tradesman.rb [hunk 1]
```diff
@@ -169,6 +171,10 @@ def job_board_key
 
     private
 
+    def queue_metrics_in_play?
```
Comment (emschwar) [inline lib/worker_roulette/tradesman.rb]: This seems a bit suspect to me. If we want to turn on queue depth tracking, I think we should just pass in a `:track_queue_depth => true` into the tradesman and have the `WorkerRoulette` send that in the correct case. Ideally, I'd like to see us passing that options directly into the WR initializer.

File: lib/worker_roulette/queue_depth.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    include ::QueueMetricTracker
+    attr_reader :channel
+
+    def initialize(channel)
+      @channel = channel
+      @queue_depth_sum = @queue_depth_count = 0
+    end
+
+    def monitor(_sender, work_orders, remaining)
+      batch_size = work_orders.length
```
Comment (emschwar) [inline lib/worker_roulette/queue_depth.rb]: It seems a bit strange to track batch size in a class called `QueueDepth`. I'd have expected them to be separate classes.

File: lib/worker_roulette.rb [hunk 3]
```diff
@@ -67,21 +67,22 @@ def initialize(config = {})
...
-          logstash_server_ip: ip_address(config[:logstash_server_name]),
-          logstash_port: config[:logstash_port]
+          metric_host: config[:metric_host],
+          metric_host_port: config[:metric_host_port],
+          metrics: config[:metrics]
         }
       )
 
-      preprocessors << QueueLatencyTracker
+      preprocessors << QueueLatency
```
Comment (emschwar) [inline lib/worker_roulette.rb]: Not to be a pedantic jerk or anything, but maybe we ought to fold this into the `QueueMetricTracker`? It is a queue metric, after all. :-\

**Comment by emschwar** [comment]
Seems legit. :shipit: (unless you want to address my pedantically jerkful comment)

File: lib/worker_roulette.rb (hunk 3 above)
Comment (james-barnard) [inline lib/worker_roulette.rb]: The queue_latency tradesman does include QueueMetricTracker and takes advantage of all it's goodness. QueueLatency classes are preprocessors also so they are called per message, not per fetch.

**Pull request merged**